caip-browser_extension_provider_communication.md

```diff
@@ -0,0 +1,141 @@
+Web extensions today will typically inject a JavaScript provider API into websites as a global variable. For example, in the Ethereum ecosystem this provider API is standardized in EIP-1193, and is conventionally injected as `window.ethereum`.
+
+This injected API strategy has some advantages:
+* For websites developers, using a global variable is simple and requires no effort on their part to setup.
+* It allows websites to support any web extension following this standard with no additional effort.
+
+However, the injected API strategy has many disadvantages:
+* It depends upon the web extension having read and write access to every website, which is a scary permission that web extension authors might otherwise be able to avoid asking for.
+* It slows down every website by injecting additional code to be parsed and executed. It even slows down the initial page load in most cases, because many web extensions inject the provider synchronously to maintain compatibility with websites that expect it to be available immediately.
+* Injecting code into a website can break it.
+* It provides no way for web extension authors to safely make breaking changes to their provider API, resulting in even more code being injected to support legacy APIs.
+
+An alternative strategy would be for a website to embed its own provider. A provider could be offered as a library, to be embedded by the website author. This strategy can address all disadvantages of the injected provider approach:
+* If this strategy became widespread enough, it would allow some web extensions to stop asking for write access to all pages.
+  * NOTE: Should mention discovery somewhere since that will require injection still
+* The website author can control when the provider is initialized, and fine-tune performance.
+* No code needs to be injected, so websites would no longer be broken by injected code.
+* The provider library can be published with breaking changes, allowing changes to the API without needing to embed legacy code in each website.
+
+A provider library can be similarly easy to use for website authors as well, requiring nothing more than a single script tag to import a library and get an equivalent experience to using an the injected provider.
+
+Web extension inter-operability is a challenge for embeded providers though. That is what this proposal means to address. Today there is no way to write a provider such that it is compatible with any web extension. Web extensions differ today in how they communicate with wallets, from the messaging system used to the messaging format. These details often aren't publicly documented or treated as a public-facing API, so they can change without notice, making it risky even to embed support for popular conventions used today.
+
+A standard method for providers to communicate with web extebsions would allow website authors to embed their own providers without losing web extension inter-operability.
```

Comment:
```suggestion
A standard method for providers to communicate with web extensions would allow website authors to embed their own providers without losing web extension inter-operability.
```